### PR TITLE
Add JSON-gRPC transcoding filter

### PR DESCRIPTION
**Pull request by lizan**
Initial version of transcoding filter (#163), will have follow up PRs for:
- Better error handling
- Tests
- Docs

### CODE REVIEW

**Comment by lizan** [comment]
cc @gurgenh @rkpagadala

File: src/envoy/transcoding/config.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  return Status::OK;
+}
+
+Status Config::MethodToRequestInfo(
+    const google::protobuf::MethodDescriptor* method,
+    google::api_manager::transcoding::RequestInfo* info) {
+  // TODO: support variable bindings
+
+  auto request_type_url =
+      kTypeUrlPrefix + "/" + method->input_type()->full_name();
```
Comment (qiwzhang) [inline src/envoy/transcoding/config.cc]: so response_type_url is the same as request_type_url?

If so, can the code be shared?

**Comment by mattklein123** [comment]
@lizan Are we going to eventually move this to Envoy repo so that everyone can use it?

**Comment by lizan** [comment]
@mattklein123 Yes, I will merge here first as this is not complete yet, and #228 blocks moving the work to envoy.

File: src/envoy/transcoding/config.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ private:
+  std::unique_ptr<JsonRequestTranslator> request_translator_;
+  std::unique_ptr<ResponseToJsonTranslator> response_translator_;
+  std::unique_ptr<TranscoderInputStream> request_stream_;
+  std::unique_ptr<TranscoderInputStream> response_stream_;
+};
+}
+
+Config::Config(const Json::Object& config, Server::Instance& server) {
+  std::string proto_descriptor_file = config.getString("proto_descriptor");
```
Comment (lizan) [inline src/envoy/transcoding/config.cc]: That is expected, if you specified transcoding filter but you don't have proto_descriptor, it should crash.

**Pull request merged**